agent/engine/engine_integ_test.go

```diff
@@ -600,6 +613,7 @@ func testNetworkMode(t *testing.T, networkMode string) {
 	assert.Equal(t, networkMode, networks[0], "did not find the expected network mode")
 
 	// Kill the existing container now
+	wg.Wait()
```

Comment:
AddTask() isn't the only place the data gets modified-- it kicks off a routine in the background that continues to modify its own state. 

Another approach you could take here would be to lock the state of the task before copying it:

```
testTask.SomeRLockFunction()
taskUpdate := *testTask
testTask.SomeRUnlockFunction()

Reply:
👍Im going to test some other ways and create a new PR.